api/data_refinery_api/views.py

```diff
@@ -485,3 +490,45 @@ def get(self, request, format=None):
         data['processor_jobs']['average_time'] = ProcessorJob.objects.filter(start_time__isnull=False, end_time__isnull=False).aggregate(average_time=Avg(F('end_time') - F('start_time')))['average_time']
 
         return Response(data)
+
+###
+# Transcriptome Indices
+###
+
+class TranscriptomeIndexDetail(APIView):
+    """
+    Retrieve the S3 URL to an OrganismIndex.
+    """
+    def get_s3_url(self, organism_name, transcription_length):
+        """
+        Gets the S3 url associated with the organism and length.
+        Organism must be specified in underscore-delimited uppercase, i.e. "GALLUS_GALLUS".
+        Length must either be "long" or "short"
+        """
+
+        if ORGANISM_INDEX_BUCKET == "NO_BUCKET":
+            raise Http404
+
+        transcription_length = "TRANSCRIPTOME_" + transcription_length.upper()
+
+        try:
+            organism_index = (OrganismIndex.public_objects.exclude(s3_key__exact="")
+                              .distinct("organism__name", "index_type")
+                              .get(organism__name=organism_name, index_type=transcription_length))
+            return "%s.s3.amazonaws.com/%s" % (ORGANISM_INDEX_BUCKET, organism_index.s3_key)
+        except OrganismIndex.DoesNotExist:
+            raise Http404
+
+    def get(self, request, format=None):
+        params = request.query_params
+        errors = dict()
+        if "organism" not in params:
+            errors["organism"] = "You must specify the organism of the index you want"
+        if "length" not in params:
+            errors["length"] = "You must specify the length of the transcriptome index"
+
+        if len(errors) > 0:
+            raise ValidationError(errors)
+
+        url = self.get_s3_url(params["organism"], params["length"])
+        return Response(url)
```

Comment:
If you make the s3_url a property of the Index, and we add a proper serializer, then we can serialize the entire object (or alternately make this a sub feature of the Organism itself). It seems a bit unusual for one of our endpoints to only return a single url value with no other information about the object.